### DISCUSSION TITLE
AVLN3000: Can't use resource for Foreground property of TextPresenter

### DISCUSSION THREAD

**Discussion by styris-ame** [AvaloniaUI/Avalonia#15691] [Q&A] [answered]
### Describe the bug

See the following XAML using a TextPresenter:

```XAML
<TextPresenter Text="Test" Foreground="{DynamicResource TextPrimaryBrush}"/>
```

It gives me this error upon compilation:
`Error AVLN3000 : Unable to find suitable setter or adder for property Foreground of type Avalonia.Controls:Avalonia.Controls.Presenters.TextPresenter for argument Avalonia.Base:Avalonia.Data.IBinding, available setter parameter lists are:
Avalonia.Media.IBrush Line 68, position 97.`

However I can still set the brush like this:
```XAML
<TextPresenter Text="Test">
    <TextPresenter.Foreground>
        <SolidColorBrush Color="{DynamicResource TextPrimaryColor}"/>
    </TextPresenter.Foreground>
</TextPresenter>
```

The following works too:

```XAML
<TextPresenter Text="Test" Foreground="Red"/>
```

I don't think this matters, but this is my brush resource:
```XAML
<SolidColorBrush x:Key="TextPrimaryBrush" Color="#191a1a"/>
```

### To Reproduce

Self explanatory

### Avalonia version

11.1.0-beta2

### OS

Windows

### Additional context

I tested it on another project too, with the same result.

**Comment by maxkatz6** [comment, marked as answer]
You can't bind to TextPresenter properties. This is a presenter control, designed to be used as a building block of more complex controls (like, TextBox).
